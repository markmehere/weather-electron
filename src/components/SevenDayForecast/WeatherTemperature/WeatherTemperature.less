.max {
  color: #eee;
}

.min {
  color: #aaa;
}
