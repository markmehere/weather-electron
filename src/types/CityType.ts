export type CityType = {
  value: number; /* woeid */
  label: string; /* city, country code (e.g. Melbourne, AU) */
};