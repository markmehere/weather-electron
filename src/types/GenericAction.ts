import { Action } from "redux";

export interface GenericAction<T> extends Action {
  payload: T;
} 
