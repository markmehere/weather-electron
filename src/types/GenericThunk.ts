import { ThunkAction } from "redux-thunk";
import { AppStore } from "../redux/reducers";

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type GenericThunk = ThunkAction<any, AppStore, undefined, any>;
