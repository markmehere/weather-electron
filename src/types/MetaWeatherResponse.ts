import { WeatherKind } from "./WeatherKind";

export interface ConsolidatedWeatherResponse {
  id: string;
  weather_state_name: string;
  weather_state_abbr: WeatherKind;
  created: string;
  applicable_date: string;
  min_temp: number;
  max_temp: number;
  wind_speed: number;
  wind_direction: number;
  air_pressure: number;
  humidity: number;
  visibility: number;
  predictability: number;
}

export interface MetaWeatherResponse {
  consolidated_weather: ConsolidatedWeatherResponse[]
}