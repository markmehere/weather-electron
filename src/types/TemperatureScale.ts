export enum TemperatureScale {
  Celsius,
  Fahrenheit
}
