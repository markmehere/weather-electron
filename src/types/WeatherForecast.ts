import { WeatherKind } from "./WeatherKind";

export interface WeatherForecast {
  max: number;
  min: number;
  date: string;
  kind: WeatherKind;
}
