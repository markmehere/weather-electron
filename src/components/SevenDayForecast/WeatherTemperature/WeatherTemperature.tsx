import React from "react";
import styles from "./WeatherTemperature.less";
import { TemperatureScale } from "../../../types/TemperatureScale";

export interface WeatherTemperatureProps {

  // The maximum in centigrade
  max: number;

  // The minimum in centrigrade
  min: number;

  // The temperature scale to use
  scale: TemperatureScale;

}

export class WeatherTemperature extends React.Component<WeatherTemperatureProps> {

  convert = (value: number) => (value * 1.8 + 32.0).toFixed(0);

  render() {
    const max = (this.props.scale === TemperatureScale.Celsius) ? this.props.max.toFixed(0) : this.convert(this.props.max);
    const min = (this.props.scale === TemperatureScale.Celsius) ? this.props.min.toFixed(0) : this.convert(this.props.min);
    const unit = (this.props.scale === TemperatureScale.Celsius) ? 'C' : 'F';

    return <div className={styles.weatherTemperature}>
      <span className={styles.max}>{max}&deg;{unit}</span> / <span className={styles.min}>{min}&deg;{unit}</span>
    </div>;
  }

}