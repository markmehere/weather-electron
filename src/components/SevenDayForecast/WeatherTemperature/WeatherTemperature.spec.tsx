import React from 'react';
import { shallow } from 'enzyme';
import { WeatherTemperature } from './WeatherTemperature';
import { TemperatureScale } from '../../../types/TemperatureScale';

describe('<WeatherTemperature />', () => {

  const defaultProps = {
    max: 20.2,
    min: -2.2,
    scale: TemperatureScale.Celsius
  };
  
  const defaultPropsFaren = {
    ...defaultProps,
    scale: TemperatureScale.Fahrenheit
  };

  it('should convert to farenheit correctly', () => {
    const mockWeatherSymbol = new WeatherTemperature(defaultProps);
    expect(mockWeatherSymbol.convert(30.2)).toBe('86');
    expect(mockWeatherSymbol.convert(28.9)).toBe('84');
    expect(mockWeatherSymbol.convert(-5.45)).toBe('22');
  });

  it('should render celsius to integer value only', () => {
    const result = shallow(<WeatherTemperature {...defaultProps}  />);
    expect(result.text()).toContain('20\u00B0C / -2\u00B0C');
  });

  it('matches snapshot with celsius scale', () => {
    const result = shallow(<WeatherTemperature {...defaultProps}  />);
    expect(result).toMatchSnapshot();
  });

  it('matches snapshot with farenheit scale', () => {
    const result = shallow(<WeatherTemperature {...defaultPropsFaren}  />);
    expect(result).toMatchSnapshot();
  });

});
