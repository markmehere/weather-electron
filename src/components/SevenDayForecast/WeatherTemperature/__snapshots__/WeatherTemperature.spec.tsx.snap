// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherTemperature /> matches snapshot with celsius scale 1`] = `
<div
  className="weatherTemperature"
>
  <span
    className="max"
  >
    20
    째
    C
  </span>
   / 
  <span
    className="min"
  >
    -2
    째
    C
  </span>
</div>
`;

exports[`<WeatherTemperature /> matches snapshot with farenheit scale 1`] = `
<div
  className="weatherTemperature"
>
  <span
    className="max"
  >
    68
    째
    F
  </span>
   / 
  <span
    className="min"
  >
    28
    째
    F
  </span>
</div>
`;
